### PR TITLE
vmctl: init at 1.93.1

### PR DESCRIPTION
**Pull request by viktornordling**
## Description of changes

Add vmctl.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nagy — CHANGES_REQUESTED** [review]
Thanks, can you follow the commit message conventions at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions ?

**Comment by viktornordling** [comment]
> Thanks, can you follow the commit message conventions at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions ?

Done!

File: pkgs/servers/monitoring/vmctl/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  vendorHash = null;
+
+  subPackages = ["app/vmctl"];
+
+  meta = with lib; {
+    homepage = "https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/app/vmctl";
+    description = "VictoriaMetrics command-line tool";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [nullx76];
```
Comment (nagy) [inline pkgs/servers/monitoring/vmctl/default.nix:28]: @viktornordling  it seems like you have entered a different maintainer than you. Or is this an alternative account of yours? @NULLx76  is this okay with you to take maintainership of this package ?

Comment (NULLx76) [reply pkgs/servers/monitoring/vmctl/default.nix:28]: I wouldn't mind being co-maintainer, but I do think it's nice if @viktornordling also lists themselves as a maintainer.

Comment (viktornordling) [reply pkgs/servers/monitoring/vmctl/default.nix:28]: Yeah, sorry, I assumed the maintainer listed would be someone from VictoriaMetrics, but maybe that was silly of me. :) I don't mind being listed.

Comment (viktornordling) [reply pkgs/servers/monitoring/vmctl/default.nix]: Added myself.

Comment (viktornordling) [reply pkgs/servers/monitoring/vmctl/default.nix]: Sorry, I didn't realize I need to add myself there as well. Done.

**Comment by nagy** [comment]
@viktornordling the part of the `maintainer-list.nix` file needs to be in a separate commit. You can find good examples of the commit message for that here: https://github.com/NixOS/nixpkgs/commits/master/maintainers/maintainer-list.nix

**Comment by viktornordling** [comment]
> 

Created a separate PR here: https://github.com/NixOS/nixpkgs/pull/252676

**Comment by nagy** [comment]
You could have done it in this PR as well, but this is ok also. All that's left here is to remove the maintainer lines from this commit, then it is good.

**Comment by viktornordling** [comment]
> You could have done it in this PR as well, but this is ok also. All that's left here is to remove the maintainer lines from this commit, then it is good.

My bad again. Removed it from this PR now.

**Review by nagy — APPROVED** [review]

**Review by NULLx76 — APPROVED** [review]

**Comment by marsam** [comment]
`vmctl` is already packaged as part of `victoriametrics`. Is there a reason to have it in a separated package?

```
📂 /nix/store/ycw8pvyw37bbppaizkdry4qbkkyi54ry-VictoriaMetrics-1.93.1
└── 📂 bin
    ├── generator
    ├── victoria-logs
    ├── victoria-metrics
    ├── vmagent
    ├── vmalert
    ├── vmauth
    ├── vmbackup
    ├── vmctl
    └── vmrestore
```

**Comment by viktornordling** [comment]
🤦 . Because I saw that `vmagent` was its own package, I tried to ad `vmctl` as a package and noticed it didn't exist, but I didn't realize it was a part of `victoriametrics` already.

**Pull request closed without merging**